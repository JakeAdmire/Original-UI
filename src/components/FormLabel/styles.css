.formLabel {
  font-weight: 300;
  display: flex;
}