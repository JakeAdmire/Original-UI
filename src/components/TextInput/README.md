# TextInput

Need to make a text input of some kind? An easily style-able search bar, or some login fields? **TextInput** is your component! Simply import `TextInput` and use the [parameters](#parameters) to start customizing, or you can copy and paste directly from one of my examples below!

## Example Usage:

### Simple Search Bar:
```javascript
<TextInput
	placeholderText='Start typing to search...'
	onChange={this.handleSearchChange}
	leftIcon={<Icon iconName={'search'}/>}/>
```
**Result:**

<img width="838" alt="Screen Shot 2019-10-21 at 2 49 58 PM" src="https://user-images.githubusercontent.com/44077214/67242483-dc0dee80-f412-11e9-9d99-982c055e2a28.png">

---


### Password input with hidden value toggle:
```javascript
import { TextInput } from  'original-ui'

<TextInput 
  placeholderText='ex. "thoughtfulpassword123..."'
  labelText='Enter your password:' 
  isPassInput={this.state.isTextHidden}
  rightIcon={
    <Icon 
      iconName={this.state.isTextHidden ? 'eye' : 'eye-slash'}
      iconSize='24px'
      onPress={this.toggleHiddenText}/>
  }/>
```

**Result:**

<img width="838" alt="Screen Shot 2019-10-20 at 9 14 45 PM" src="https://user-images.githubusercontent.com/44077214/67174278-b931fa80-f37e-11e9-8e9c-fe374a3642cb.png">


## Parameters:

|Parameter| Data Type | Default Value | Description |
|--|--|--|--|
| `placeholderText` | `String` | `null` | _Pass a string to display as a placeholder_ |
| labelText | `String` | `null` | _Pass a string to display as a label_ |
| labelStyle | `Object` | `{ color: 'orange' }` | _Pass a style object to set the label's style rules_ |
| isPassInput | `Boolean` | `false` | _Pass a boolean to determine whether you want the input's text obscured_ |
| leftIcon | `Component` | `null` | _Pass an [Icon](#icon) component to display on the left side of the input_ |
| rightIcon | `Component` | `null` | _Pass an [Icon](#icon) component to display on the right side of the input_ |
| `width` | `String` | `'100%'` | _Pass a string to constrict the component's width_ |
| `height` | `String` | `'50px'` | _Pass a string to constrict the component's height_ |
| `textStyle` | `Object` | `{ fontSize: '16px', color: 'grey' }` | _Pass a style object to set the input text's style rules_ |
| `onChange` | `Function` | `null` | _Pass a method that will receive the `event` object as a parameter_ |
| `value` | 'String' | `null` | _Pass a string to set the input's value_ |
---