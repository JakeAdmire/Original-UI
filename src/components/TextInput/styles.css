.inputlabel {
  font-weight: 300;
}

.inputwrapper {
  box-sizing: border-box;
  padding: 6px;
  display: flex;
  align-items: center;
  transition: all .3s ease;
  background: #e7e7e7;
  border-radius: 50px;
}

.textinput {
  box-sizing: border-box;
  padding: 0 10px;
  outline: none;
  width: 100%;
  height: 100%;
  border: none;
  background: none;
  transition: all .3s ease;
}

.textinput::placeholder {
  color: #00000033;
}

.inputshadow {
  box-shadow: inset 0 0 12px 0px rgba(0, 0, 0, 0.15);
}