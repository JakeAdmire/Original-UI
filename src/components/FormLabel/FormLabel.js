import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

export function FormLabel({
  context = null,
  labelPlacement = 'top',
  labelStyle = { color: 'orange' },
  labelText = null
}) {
  const placeLabel = () => {
    if (labelPlacement === 'top') labelStyle['flexDirection'] = 'column'
    if (labelPlacement === 'right') labelStyle['flexDirection'] = 'row-reverse'
    if (labelPlacement === 'bottom') labelStyle['flexDirection'] = 'column-reverse'
    if (labelPlacement === 'left') labelStyle['flexDirection'] = 'row'
    return labelStyle
  }

  return (
    <div>
      <label
        className={styles.formLabel}
        style={placeLabel()}
      >
        {labelText}
        {context}
      </label>
    </div>
  )
}

FormLabel.propTypes = {
  context: PropTypes.element,
  labelPlacement: PropTypes.string,
  labelStyle: PropTypes.object,
  labelText: PropTypes.string
}
