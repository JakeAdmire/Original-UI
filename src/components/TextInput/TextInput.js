import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

export class TextInput extends Component {
  static propTypes = {
    placeholder: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
    fontsize: PropTypes.string,
    showShadow: PropTypes.bool,
    border: PropTypes.string,
    radius: PropTypes.string,
    vanishingPlaceholder: PropTypes.bool,
    changeMethod: PropTypes.func
  };

  render() {
    const {
      placeholder,
      width = '100%',
      height = '50px',
      fontsize = '16px',
      showShadow = true,
      border = 'orangered',
      radius = '10px',
      vanishingPlaceholder = true,
      changeMethod
    } = this.props

    let inlineStyles = {
      width: width,
      height: height,
      fontSize: fontsize,
      borderColor: border,
      borderRadius: radius
    }

    let inputClass = `
      ${styles.textinput} 
      ${showShadow && styles.inputshadow}
      ${vanishingPlaceholder && styles.vanishingPlaceholder}
    `

    return (
      <input
        className={inputClass}
        placeholder={placeholder || null}
        style={inlineStyles}
        onChange={changeMethod !== null ? (e) => changeMethod(e) : null}
      />
    )
  }
}
