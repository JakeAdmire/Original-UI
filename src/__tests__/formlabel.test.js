/* eslint-disable padded-blocks */
import React from 'react'
import { shallow } from 'enzyme'

import { FormLabel } from '../components/FormLabel/FormLabel'
import { TextInput } from '../components/TextInput/TextInput'

describe('FormLabel', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <FormLabel />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('[Parameter] context', () => {

    it('should not have a default value', () => {
      let context = wrapper.find('label').props().children[1]
      expect(context).toEqual(null)
    })

    it('should give FormLabel child element reflecting parameter', () => {
      expect(wrapper.find('label').props().children[1]).toEqual(null)

      let input = <TextInput />
      wrapper.setProps({ context: input })

      expect(wrapper.find('label').props().children[1]).toEqual(input)
    })

  })

  describe('[Parameter] labelPlacement', () => {

    it('should have a default value', () => {
      let label = wrapper.find('label')
      expect(label.props().style.flexDirection).toEqual('column')
    })

    it('["right"] should give FormLabel style reflecting parameter', () => {
      wrapper.setProps({ labelPlacement: 'right' })

      let label = wrapper.find('label')
      expect(label.props().style.flexDirection).toEqual('row-reverse')
    })

    it('["bottom"] should give FormLabel style reflecting parameter', () => {
      wrapper.setProps({ labelPlacement: 'bottom' })

      let label = wrapper.find('label')
      expect(label.props().style.flexDirection).toEqual('column-reverse')
    })

    it('["left"] should give FormLabel style reflecting parameter', () => {
      wrapper.setProps({ labelPlacement: 'left' })

      let label = wrapper.find('label')
      expect(label.props().style.flexDirection).toEqual('row')
    })

  })

  describe('[Parameter] labelStyle', () => {

    it('should have a default value', () => {
      let label = wrapper.find('label')
      expect(label.props().style.color).toEqual('orange')
    })

    it('should give FormLabel style reflecting parameter', () => {

    })

  })

  describe('[Parameter] labelText', () => {

    it('should not have a default value', () => {

    })

    it('should give FormLabel innerText reflecting parameter', () => {

    })

  })

})
