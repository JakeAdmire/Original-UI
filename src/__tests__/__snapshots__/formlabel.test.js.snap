// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLabel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className={undefined}
        style={
                Object {
                        "color": "orange",
                        "flexDirection": "column",
                      }
        }
>
        
</label>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": undefined,
        "style": Object {
          "color": "orange",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className={undefined}
          style={
                    Object {
                              "color": "orange",
                              "flexDirection": "column",
                            }
          }
>
          
</label>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            null,
          ],
          "className": undefined,
          "style": Object {
            "color": "orange",
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Array [
          null,
          null,
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
